From 8a33204ec8502041d01c593f8dba786a309f00f4 Mon Sep 17 00:00:00 2001
From: Michael D Labriola <mdlabriola@yahoo.com>
Date: Tue, 15 Feb 2011 16:42:58 -0500
Subject: [PATCH] m4: fixed some "AC_LANG_CONFTEST: no AC_LANG_SOURCE" warnings

This commit fixes some autoconf warnings by inserting AC_LANG_SOURCE and
AC_LANG_PROGRAM macros where needed.
---
 m4/ac_check_ciao.m4       |    4 ++--
 m4/ac_check_swi_prolog.m4 |    4 ++--
 m4/ac_prog_javah.m4       |    4 ++--
 m4/libtool.m4             |    2 +-
 4 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/m4/ac_check_ciao.m4 b/m4/ac_check_ciao.m4
index 78dc1cf..629f539 100644
--- a/m4/ac_check_ciao.m4
+++ b/m4/ac_check_ciao.m4
@@ -55,13 +55,13 @@ _CIAO_PRINT_INCLUDE_DIR_EOF
   AC_LANG_PUSH(C++)
   AC_CHECK_HEADER(ciao_prolog.h,
                   AC_MSG_CHECKING([for Ciao Prolog version x.y.z or later])
-                  AC_COMPILE_IFELSE(
+                  AC_COMPILE_IFELSE([AC_LANG_SOURCE(
 [#include <ciao_prolog.h>
 
 int
 main() {
   return 0;
-}],
+}])],
                     AC_MSG_RESULT(yes),
                     ciao_prolog=""
                     AC_MSG_RESULT(no)
diff --git a/m4/ac_check_swi_prolog.m4 b/m4/ac_check_swi_prolog.m4
index 4ca9955..949a95c 100644
--- a/m4/ac_check_swi_prolog.m4
+++ b/m4/ac_check_swi_prolog.m4
@@ -54,7 +54,7 @@ then
   # We require SWI-Prolog 5.6.0 or later.
   AC_CHECK_HEADER(SWI-Prolog.h,
                   AC_MSG_CHECKING([for SWI-Prolog version 5.6.0 or later])
-                  AC_COMPILE_IFELSE(
+                  AC_COMPILE_IFELSE([AC_LANG_SOURCE(
 [#include <SWI-Prolog.h>
 
 int
@@ -65,7 +65,7 @@ main() {
 
   ;
   return 0;
-}],
+}])],
                     AC_MSG_RESULT(yes),
                     swi_prolog=""
                     AC_MSG_RESULT(no)
diff --git a/m4/ac_prog_javah.m4 b/m4/ac_prog_javah.m4
index 6886a2d..b6b3063 100644
--- a/m4/ac_prog_javah.m4
+++ b/m4/ac_prog_javah.m4
@@ -33,7 +33,7 @@ else
 fi
 
 if test x"`eval 'echo $ac_cv_path_JAVAH'`" != xno ; then
-  AC_PREPROC_IFELSE([#include <jni.h> ] ,,[
+  AC_PREPROC_IFELSE([AC_LANG_PROGRAM([#include <jni.h> ])] ,,[
     ac_save_CPPFLAGS="$CPPFLAGS"
 changequote(, )dnl
     ac_dir=`echo $ac_cv_path_JAVAH | sed 's,\(.*\)/[^/]*/[^/]*$,\1/include,'`
@@ -42,7 +42,7 @@ changequote([, ])dnl
     JNIFLAGS="-I$ac_dir -I$ac_dir/$ac_machdep"
     CPPFLAGS="$ac_save_CPPFLAGS $JNIFLAGS"
     AC_SUBST(JNIFLAGS)
-    AC_PREPROC_IFELSE([#include <jni.h>],
+    AC_PREPROC_IFELSE([AC_LANG_PROGRAM([#include <jni.h>])],
                CPPFLAGS="$ac_save_CPPFLAGS",
                AC_MSG_WARN([unable to include <jni.h>])
 	       JAVAH=no)
diff --git a/m4/libtool.m4 b/m4/libtool.m4
index 39ba996..d24bd05 100644
--- a/m4/libtool.m4
+++ b/m4/libtool.m4
@@ -4826,7 +4826,7 @@ _LT_EOF
 	# implicitly export all symbols.
         save_LDFLAGS="$LDFLAGS"
         LDFLAGS="$LDFLAGS -shared ${wl}-exported_symbol ${wl}foo ${wl}-update_registry ${wl}/dev/null"
-        AC_LINK_IFELSE(int foo(void) {},
+        AC_LINK_IFELSE([AC_LANG_SOURCE([int foo(void) {}])],
           _LT_TAGVAR(archive_expsym_cmds, $1)='$CC -shared $libobjs $deplibs $compiler_flags ${wl}-soname ${wl}$soname `test -n "$verstring" && $ECHO "X${wl}-set_version ${wl}$verstring" | $Xsed` ${wl}-update_registry ${wl}${output_objdir}/so_locations ${wl}-exports_file ${wl}$export_symbols -o $lib'
         )
         LDFLAGS="$save_LDFLAGS"
-- 
1.7.1.1

